src/core/schema.js

```diff
@@ -75,6 +76,18 @@ function processPropertyDefinition (propDefinition) {
   // Fill in default value.
   if (!('default' in propDefinition)) {
     propDefinition.default = propType.default;
+  } else if (typeName === 'vec2' && typeof defaultVal === 'string') {
```

Comment:
Yeah I'm going to make a function that does this. I'm make it so that if the type is substantiated the default value has to be the same as well. For example if `type` is `number` `default` should also be a `number`. Sound good?